### PR TITLE
Update jib-core default application cache dir

### PR DESCRIPTION
**Pull request by loosebazooka**
- in <system-tmp>/jib-core-application-layer-cache
- does not delete on exit (should help speed on rebuilds)
- shared across all builds that do not specify a custom
  application cache directory

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java [hunk 1]
```diff
@@ -312,11 +318,27 @@ Path getBaseImageLayersCacheDirectory() {
...
-        applicationLayersCacheDirectory.toFile().deleteOnExit();
+        Path systemTmp = tmpDirSupplier.get();
+        Path jibTempCache = systemTmp.resolve(DEFAULT_APPLICATION_CACHE_DIRECTORY_NAME);
+        if (Files.exists(jibTempCache)) {
+          if (!Files.isDirectory(jibTempCache)) {
+            throw new CacheDirectoryCreationException(
+                jibTempCache.toString() + " already exists and is not a directory");
+          }
+        } else {
+          Files.createDirectory(jibTempCache);
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java:339]: Actually think I can clean up all this logic and just use Files.createDirectories ? Or just lets the cache writer handle this?

File: jib-core/src/test/java/com/google/cloud/tools/jib/api/ContainerizerTest.java [hunk 2]
```diff
@@ -19,16 +19,23 @@
...
 import org.junit.Assert;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 import org.mockito.Mockito;
 
 /** Tests for {@link Containerizer}. */
 public class ContainerizerTest {
 
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
```
Comment (chanseokoh) [inline jib-core/src/test/java/com/google/cloud/tools/jib/api/ContainerizerTest.java:37]: `final`

File: jib-core/src/test/java/com/google/cloud/tools/jib/api/ContainerizerTest.java [hunk 3]
```diff
@@ -121,4 +128,15 @@ public void testGetImageConfiguration_tarImage() throws InvalidImageReferenceExc
...
   }
+
+  @Test
+  public void testGetApplicationLayersCacheDirectory_defaults()
+      throws InvalidImageReferenceException, CacheDirectoryCreationException, IOException {
+    Containerizer containerizer = Containerizer.to(RegistryImage.named("registry/image"));
+    Path applicationLayersCache = containerizer.getApplicationLayersCacheDirectory();
+    Path expectedCacheDir =
+        Paths.get(System.getProperty("java.io.tmpdir"))
+            .resolve(Containerizer.DEFAULT_APPLICATION_CACHE_DIRECTORY_NAME);
```
Comment (chanseokoh) [inline jib-core/src/test/java/com/google/cloud/tools/jib/api/ContainerizerTest.java:139]: How about verifying the name in the test?
```suggestion
            .resolve("jib-core-application-layers-cache");
```

**Review by chanseokoh — APPROVED** [review]

File: jib-core/src/test/java/com/google/cloud/tools/jib/api/ContainerizerTest.java (hunk 3 above)
Comment (loosebazooka) [reply jib-core/src/test/java/com/google/cloud/tools/jib/api/ContainerizerTest.java:135]: oh right.

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/CacheDirectoryCreationException.java [hunk 4]
```diff
@@ -24,4 +24,8 @@
   public CacheDirectoryCreationException(Throwable cause) {
     super(MESSAGE, cause);
   }
+
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/CacheDirectoryCreationException.java:27]: actually this change is now unnecessary

**Review by chanseokoh — APPROVED** [review]

**Pull request merged**